src/test/java/com/mkdk72ki/asgmt10/integrationtest/UserRestApiIntegrationTest.java

```diff
@@ -0,0 +1,352 @@
+
+    }
+
+    @Test
+    @DataSet(value = "datasets/users.yml")
+    @Transactional
+    void IDで指定したユーザーが取得できること() throws Exception {
+        String response = mockMvc.perform(MockMvcRequestBuilders.get("/users/{id}", 1))
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andReturn().getResponse().getContentAsString(StandardCharsets.UTF_8);
+
+        JSONAssert.assertEquals("""
+                {
+                    "id": 1,
+                    "name": "山田太郎",
+                    "ruby": "yamada taro",
+                    "birthday": "1990-03-04",
+                    "email": "yamada@mkdk.com"
+                }
+                            """, response, JSONCompareMode.STRICT);
+
+    }
+
+    @Test
+    @DataSet(value = "datasets/users.yml")
+    @Transactional
+    void 存在しないIDを指定したときに404エラーが返されること() throws Exception {
+        String response = mockMvc.perform(MockMvcRequestBuilders.get("/users/{id}", 99))
+                .andExpect(MockMvcResultMatchers.status().isNotFound())
+                .andReturn().getResponse().getContentAsString(StandardCharsets.UTF_8);
+
+        JSONAssert.assertEquals("""
+                {
+                    "error": "Not Found",
+                    "timestamp": "2023-12-19T13:22:34.194271200+09:00[Asia/Tokyo]",
+                    "path": "/users/99",
+                    "status": "404"
+                }
+                """, response, new CustomComparator(JSONCompareMode.STRICT, new Customization("timestamp", ((o1, o2) -> true))));
+    }
+
+    // POST
+
+    @Test
+    @DataSet(value = "datasets/users.yml")
+    @ExpectedDataSet(value = "datasets/insert-users.yml", ignoreCols = "id")
+    @Transactional
+    void ユーザーが登録できること() throws Exception {
+        String response = mockMvc.perform(MockMvcRequestBuilders.post("/users")
+                        .contentType(MediaType.APPLICATION_JSON).content("""
+                                {
+                                 "name": "田村敦",
+                                 "ruby": "tamura atsushi",
+                                 "birthday": "1995-09-18",
+                                 "email": "tamura@mkdk.com"
+                                }
+                                 """)).andExpect(MockMvcResultMatchers.status().isCreated()).
+                andReturn().getResponse().getContentAsString(StandardCharsets.UTF_8);
+
+        JSONAssert.assertEquals("""
+                {
+                    "message": "登録しました"
+                }
+                            """, response, JSONCompareMode.STRICT);
+
+    }
+
+    @Test
+    @DataSet(value = "datasets/users.yml")
+    @Transactional
+    void 名前が空文字で登録したときに400エラーが返されること() throws Exception {
+        mockMvc.perform(MockMvcRequestBuilders.post("/users")
+                        .contentType(MediaType.APPLICATION_JSON).content("""
+                                {
+                                  "name": " ",

```

Comment:
これは空文字でなく半角スペースのみと表現するほうが適切ですね。
```suggestion
                                  "name": "",

```

Reply:
6ea206e3a835dde7cce9f16e980658055c395842 こちら全ての空文字の検証について修正いたしましたのでご確認お願いいたします。

Reply:
OKです！